iceoryx_utils/include/iceoryx_utils/lockfree_queue/readme.md

```diff
@@ -0,0 +1,140 @@
+# Lock Free queue 
+
+We explain some details of the lock free queue in order to provide an intuition for the way the lock free queue works. This could serve as a basis for a formal proof later if desired.
+
+## Index queue analysis
+
+In the following all numbers are natural numbers and hence bounded unsigned integers in the implementation.
+
+### Capacity and Cycle Length
+The index queue has some capacity n and stores indices in the range [0, n[.
+The number n is also called cycle length.
+
+Assume n = 4 in this explanation, so we can store the indices 0, 1, 2, 3.
+
+
+### Index representation
+Each index i corresponds to an equivalence class [i] modulo n, i.e. 
+[i] = {j | j = c*n + i, n >= 0}

```

Comment:
c is not yet defined

Reply:
I will correct this and rework the whole explanation together with the resize feature.

Reply:
should be c>=0, oh well